packages/spacecat-shared-utils/src/log-wrapper.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/**
+ * Wraps the provided function to enhance the logging functionality by including the
+ * `jobId` in each log statement.
+ *
+ * @param {function} fn - The original function that will be wrapped.
+ * @returns {function(object, object): Promise<Response>} A new function that, when invoked,
+ *  will automatically include the `jobId` in all log statements within the context.
+ */
+export function logWrapper(fn) {
+  return async (message, context) => {
+    const { log } = context;
+
+    if (log) {
+      if (!message || !message.jobId) {
+        log.debug('Missing jobId, hence it will not be included in log messages.');
+        context.contextualLog = log;
+
+        return fn(message, context);
+      }
+
+      const { jobId } = message;
+
+      // Enhance the log object to include jobId in each log statement
+      context.contextualLog = {
+        info: (...args) => {
+          log.info({ jobId, ...args });
```

Comment:
Can you share a screenshot from your testing of how the resulting log entries are rendered on Coralogix? 

Is leveraging this functionality to put together the "complete" log stream as simple as searching for `jobId=<UUID>` in Coralogix? 🤞

Reply:
It doesn’t appear as a key, but rather as jobId: 'f91afda0-afc8-467e-bfa3-fdbeba3037e8' as part of the message, which can be included in the search query.”